sql-plugin/src/main/scala/org/apache/spark/sql/rapids/mathExpressions.scala

```diff
@@ -369,6 +370,70 @@ abstract class CudfBinaryMathExpression(name: String) extends CudfBinaryExpressi
   override def dataType: DataType = DoubleType
 }
 
+abstract class GpuRoundBase(child: Expression, scale: Expression) extends GpuBinaryExpression
+  with Serializable with ImplicitCastInputTypes {
+
+  override def left: Expression = child
+  override def right: Expression = scale
+
+  def roundMode: RoundMode
+
+  override lazy val dataType: DataType = child.dataType match {
+    // if the new scale is bigger which means we are scaling up,
+    // keep the original scale as `Decimal` does
+    case DecimalType.Fixed(p, s) => DecimalType(p, if (_scale > s) s else _scale)
+    case t => t
+  }
+
+  // Avoid repeated evaluation since `scale` is a constant int,
+  // avoid unnecessary `child` evaluation in both codegen and non-codegen eval
+  // by checking if scaleV == null as well.
+  private lazy val scaleV: Any = scale match {
+    case _: GpuExpression => scale.columnarEval(null)
+    case _ => scale.eval(EmptyRow)
+  }
+  private lazy val _scale: Int = scaleV.asInstanceOf[Int]
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(NumericType, IntegerType)
+
+  override def doColumnar(val0: GpuColumnVector, val1: Scalar): ColumnVector = {
+    val scaleVal=val1.getInt
+    val scale = dataType match {
```

Comment:
This is not needed at all.  We already calculated the scale several different ways already.  You can use either `scaleVal` or `_scale`. I personally think that `scaleVal` might be preferable, but either should be fine.

Reply:
Sorry, I was using `scaleVal` to get the scale for non-decimals. The reason i had `scale` here was to get the correct scale for `decimals`. We have updated the scale in `DataType` for `DecimalType` and that is what is retrieved here. I think we cannot use `_scale` as it is. Please let me know if it looks okay.

Reply:
You are right I missed the part where the scale can depend on the child's scale too.